### DISCUSSION TITLE
Dynamically hide applications

### DISCUSSION THREAD

**Discussion by intelliant01** [ltsp/ltsp#677] [Q&A] [answered]
I have a peculiar need to have both libreoffice and openoffice in my image but make only one visible to the users in different circumstances. Have managed to workaround the issue of having both simultaneously installed in a distro. But am grappling with the visibility part.

Have tried the below technique by adding these lines to ltsp.conf and executed `ltsp initrd` and rebooted the client.
For example, when the preference is to show openoffice, `INCLUDE=openoffice`

```
[clients]
INCLUDE=openoffice
[openoffice]
POST_INIT_OPENOFFICE="office=libreoffice && cd /usr/share/applications && for i in `ls $office*.desktop`; do mv $i $i.bak; done"
[libreoffice]
POST_INIT_LIBREOFFICE="office=openoffice && cd /usr/share/applications && for i in `ls $office*.desktop`; do mv $i $i.bak; done"
```

But this is not working as expected.  The menu icons are still visible after every reboot of clients.

**Comment by alkisg** [comment, marked as answer]
Your quoting is wrong; `` ` ``, when inside `"`, is evaluated before the `cd` command, so the `ls` returns nothing. Try this:

```ini
[clients]
POST_INIT_RM_LIBREOFFICE_MENUS="rm -f /usr/share/applications/libreoffice*.desktop"
```

**Comment by intelliant01** [comment]
Thanks for pointing out the error.
